Integrations/integration-UrlScan.yml

```diff
@@ -0,0 +1,384 @@
+description: UrlScan.io reputation
+detaileddescription: |-
+  This integration checks domain information from the UrlScan.io Database.
+  No authentication is needed.
+configuration:
+- display: Url for UrlScan API
+  name: url
+  defaultvalue: https://urlscan.io/api/v1/
+  type: 0
+  required: false
+- display: API Key(this used only for submitting new urls for scan)
+  name: apikey
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    var sendRequest = function(method, arg) {
+
+        url = params.url;
+        if (url[url.length - 1] === '/') {
+            url = url.substring(0, url.length - 1);
+        }
+
+        var requestUrl = url + arg;
+        var res = http(
+            requestUrl,
+            {
+                Method: method,
+                Headers: {
+                }
+            },
+            params.proxy
+        );
+
+        if ((res.StatusCode < 200 || res.StatusCode >= 300) && res.success === false) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        try{
+            return JSON.parse(res.Body);
+        }
+        catch(exc){
+            throw "Error: " + res.Body;
+        }
+    };
+
+
+    var searchResults = function(id){
+        var res = sendRequest("GET", "/result/"+id);
+        return res;
+    };
+
+    var searchUrl = function(url) {
+        var res = sendRequest("GET","/search?q=domain:"+url);
+        var md = "";
+        var ec = {};
+        var resDis = [];
+        var dbotScore = 0;
+        md += "## UrlScan.io Reputation for "+url;
+
+        if(res.status !== 400 && res.results.length !== 0){
```

Comment:
If is different.
In the sendRequest it can get 200 and in the Body response, it can get an error.
Here I check not the HTTP status but the JSON response.